pkg/kv/kvclient/rangefeed/scanner.go

```diff
@@ -112,7 +112,11 @@ func (f *RangeFeed) getSpansToScan(ctx context.Context) (func() []roachpb.Span,
 				return err
 			}
 		}
-		_, err := frontier.Forward(sp, f.initialTimestamp)
+		advanced, err := frontier.Forward(sp, f.initialTimestamp)
```

Comment:
What do you mean "once on checkpoint" ? this _is_ us calling it on a (potential) checkpoint. We've just finished moving a span from time zero to time initial, and we're informing the frontier visitor that it could elect to visit now if it wants to see the updated frontier.

Reply:
At the time I wrote this comment, I was reading the comment for `FrontierSpanVisitor ` too literally:

> // FrontierSpanVisitor is called when the FrontierSpanVisitTrigger requests the
// frontier be visited after a checkpoint.

The comment could be revised to "requests the frontier to be visited after it is advanced". But CI is green.